sweep/aggregator.go

```diff
@@ -21,12 +23,152 @@ const (
 	DefaultFeeRateBucketSize = 10
 )
 
+// inputSet is a set of inputs that can be used as the basis to generate a tx
+// on.
+type inputSet []input.Input
+
+// Cluster defines an interface that prepares inputs of a cluster to be grouped
+// into a list of sets that can be used to create sweep transactions.
+type Cluster interface {
```

Comment:
Good question re semantic distinction here. Is it that given the specified constraints (fee rate + max inputs), some inputs in the cluster may be ineligible? 

Also does the wallet have to be passed as an input vs having it be implicitly be a part of the concrete impl of the cluster?

Reply:
This is an intermediate commit that helps with me moving parts out of the sweeper. Once the `UtxoAggregator` is deepen, this `Cluster` interface is removed.

> Think this shouldn't use the word cluster as that will eventually have a specific meaning come cluster mempool

Will address this in the final PR to avoid too many rebase conflicts.